test/suites/osconfig/clusterid.robot

```diff
@@ -83,10 +102,79 @@ Get MicroShift Cluster ID From Namespace
     Should Not Be Empty    ${clusterid}
     RETURN    ${clusterid}
 
+Extract Sos Report
+    [Documentation]    Extract Sos Report from the tar file
+    [Arguments]    ${sos_report_tarfile}
+
+    ${sos_report_dir}    ${rc}=    Execute Command
+    ...    dirname ${sos_report_tarfile}
+    ...    sudo=True    return_rc=True    return_stdout=True
+    Log    ${sos_report_dir}
+    Should Be Equal As Integers    0    ${rc}
+
+    ${rc}=    Execute Command
+    ...    tar xf ${sos_report_tarfile} -C ${sos_report_dir}
+    ...    sudo=True    return_rc=True    return_stdout=False
+    Should Be Equal As Integers    0    ${rc}
+
+    ${sos_report_untared}    ${rc}=    Execute Command
+    ...    find ${sos_report_dir} -type d -name "sosreport-*"
+    ...    sudo=True    return_rc=True    return_stdout=True
+    Should Be Equal As Integers    0    ${rc}
+
+    RETURN    ${sos_report_untared}
+
+Get MicroShift Cluster ID From Sos Report
+    [Documentation]    Read and return the Cluster ID from the kube-system namespace yaml description in the Sos Report.
+    [Arguments]    ${sos_report_tarfile}
+
+    ${sos_report_untared}=    Extract Sos Report    ${sos_report_tarfile}
+
+    ${namespace_yaml_path}=    Set Variable
+    ...    ${sos_report_untared}/sos_commands/microshift/namespaces/${CLUSTERID_NS}/${CLUSTERID_NS}.yaml
+    ${clusterid}    ${rc}=    Execute Command
+    ...    cat ${namespace_yaml_path} | sed -n 's/\\s\\suid:\\s//p'
+    ...    sudo=True    return_rc=True    return_stdout=True
+    Should Be Equal As Integers    0    ${rc}
+
+    Should Not Be Empty    ${clusterid}
+    RETURN    ${clusterid}
+
 Tamper With Cluster ID File
     [Documentation]    Append invalid characters to the cluster ID file.
-    ${stdout}    ${stderr}    ${rc}=    Execute Command
+    ${rc}=    Execute Command
     ...    sed -i '$ s/$/123/' ${CLUSTERID_FILE}
-    ...    sudo=True    return_rc=True    return_stdout=True    return_stderr=True
+    ...    sudo=True    return_rc=True    return_stdout=False
+
+    Should Be Equal As Integers    0    ${rc}
+
+Remove Cluster ID File
+    [Documentation]    Append invalid characters to the cluster ID file.
+    ${rc}=    Execute Command
+    ...    rm -rf ${CLUSTERID_FILE}
```

Comment:
We should only use `-r` option when deleting directories.
```suggestion
    ...    rm -f ${CLUSTERID_FILE}
```

Reply:
changed